CovertInfiltration/Src/CovertInfiltration/Classes/UISkyrangerExfiltrate.uc

```diff
@@ -0,0 +1,259 @@
+		Button1.SetGamepadIcon(class'UIUtilities_Input'.static.GetAdvanceButtonIcon());
+		Button1.SetX(-150.0 / 2.0);
+		Button1.SetY(-Button1.Height / 2.0);
+		Button1.DisableNavigation();
+	}
+	else
+	{
+		Button1.SetResizeToText(false);
+		Button1.InitButton('Button0', "", OnLaunchClicked);
+	}
+
+	Button2 = Spawn(class'UIButton', ButtonGroup);
+	if(`ISCONTROLLERACTIVE )
+	{
+		Button2.InitButton('Button1', "", OnCancelClicked, eUIButtonStyle_HOTLINK_BUTTON);
+		Button2.SetGamepadIcon(class'UIUtilities_Input'.static.GetBackButtonIcon());
+		Button2.SetX(-175.0 / 2.0);
+		Button2.SetY(Button2.Height / 2.0);
+		Button2.DisableNavigation();
+	}
+	else
+	{
+		Button2.SetResizeToText(false);
+		Button2.InitButton('Button1', "", OnCancelClicked);
+	}
+}
+
+simulated function RefreshNavigation()
+{
+	if(Button1.bIsVisible)
+	{
+		if(`ISCONTROLLERACTIVE == false)
+		{
+			Button1.EnableNavigation();
+		}
+	}
+	else
+	{
+		Button1.DisableNavigation();
+	}
+
+	if(Button2.bIsVisible)
+	{
+		if(`ISCONTROLLERACTIVE == false)
+		{
+			Button2.EnableNavigation();
+		}
+	}
+	else
+	{
+		Button2.DisableNavigation();
+	}
+
+	LibraryPanel.bCascadeFocus = false;
+	LibraryPanel.bCascadeSelection = false;
+	LibraryPanel.SetSelectedNavigation();
+	ButtonGroup.bCascadeFocus = false;
+	ButtonGroup.bCascadeSelection = false;
+	ButtonGroup.Navigator.LoopSelection = true;
+	ButtonGroup.SetSelectedNavigation();
+
+	if(Button1.bIsNavigable)
+	{
+		Button1.SetSelectedNavigation();
+	}
+	else if(Button2.bIsNavigable)
+	{
+		Button2.SetSelectedNavigation();
+	}
+}
+
+simulated function BuildScreen()
+{
+	local XComGameState NewGameState;
+
+	`XSTRATEGYSOUNDMGR.PlaySoundEvent("Geoscape_SkyrangerStop");
+	
+	NewGameState = class'XComGameStateContext_ChangeContainer'.static.CreateChangeState("Skyranger Aborts Event");
+	`XEVENTMGR.TriggerEvent('OnSkyrangerArrives', , , NewGameState);
+	`XCOMGAME.GameRuleset.SubmitGameState(NewGameState);
+
+	BuildSkyrangerPanel();
+	BuildOptionsPanel();
+
+	RefreshNavigation();
+	if (!Movie.IsMouseActive())
+	{
+		Navigator.Clear();
+	}
+}
+
+simulated function BuildSkyrangerPanel()
+{
+	// Send over to flash
+	LibraryPanel.MC.BeginFunctionOp("UpdateSkyrangerInfoBlade");
+	LibraryPanel.MC.QueueString(strSkyrangerExfiltrateTitle);
+	LibraryPanel.MC.QueueString(strSkyrangerExfiltrateSubtitle);
+	LibraryPanel.MC.EndOp();
+}
+
+simulated function BuildOptionsPanel()
+{
+	// Send over to flash
+	LibraryPanel.MC.BeginFunctionOp("UpdateSkyrangerButtonBlade");
+	LibraryPanel.MC.QueueString(GetOpName());
+	LibraryPanel.MC.QueueString(strBody);
+	LibraryPanel.MC.QueueString(strConfirmExfiltration);
+	LibraryPanel.MC.QueueString(strCancelExfiltration);
+	LibraryPanel.MC.EndOp();
+}
+
+//////////////////////
+/// Event Handling ///
+//////////////////////
+
+simulated function OnLaunchClicked(UIButton button)
+{
+	`HQPRES.UINarrative(XComNarrativeMoment'X2NarrativeMoments.Strategy.Avenger_Skyranger_Recalled');
```

Comment:
```uc
XComNarrativeMoment(DynamicLoadObject('X2NarrativeMoments.Strategy.Avenger_Skyranger_Recalled', class'XComNarrativeMoment'))
```

Reply:
not really sure what you mean.. robojumper suggested:
``HQPRES.UINarrative(XComNarrativeMoment(`CONTENT.RequestGameArchetype("X2NarrativeMoments.Strategy.Avenger_Skyranger_Recalled")));`
(i think..)